### PR TITLE
Update inline xml so it's deserializable

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Add some elements for sequences of children. Make everything
an element for now. Add header and footer.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@kyulee PTAL
cc @dotnet/jit-contrib

**Comment by AndyAyersMS** [comment]
Updated to make xml finalization properly conditional.

**Comment by kyulee1** [comment]
LGTM

File: src/jit/inline.cpp [hunk 1]
```diff
@@ -1265,6 +1275,15 @@ void InlineStrategy::DumpXml(FILE* file, unsigned indent)
...
 
+    // Dump header
+    if (!s_DumpXmlHeader)
+    {
+        fprintf(file, "<?xml version=\"1.0\"?>\n");
+        fprintf(file, "<InlineForest>\n");
+        fprintf(file, "<Methods>\n");
+        s_DumpXmlHeader = true;
+    }
+
```
Comment (CarolEidt) [inline src/jit/inline.cpp]: Minor point, but "DumpXmlHeader sounds like an imperative - like it should be dumped. Maybe s_DumpedXmlHeader?

**Comment by CarolEidt** [comment]
LGTM with one possible suggestion

**Comment by AndyAyersMS** [comment]
@CarolEidt I have another change coming in this area so I'll take up your suggestion there.

**Comment by AndyAyersMS** [comment]
@mmitche: Ubuntu failure looks like the same kind of thing @janvorli saw -- about 20 tests try to load a windows API dll.
```
Unhandled Exception: System.TypeInitializationException: The type initializer for 'System.Diagnostics.Stopwatch' threw an exception. ---> 
System.DllNotFoundException: Unable to load DLL 'api-ms-win-core-profile-l1-1-0.dll': 
The specified module could not be found. (Exception from HRESULT: 0x8007007E) 
at Interop.mincore.QueryPerformanceFrequency(Int64& value)
at System.Diagnostics.Stopwatch..cctor()
```
Haven't checked but I wonder if the the tests that fail are the ones using stopwatch...?

**Comment by AndyAyersMS** [comment]
Am going to ignore the CI issue since it's almost certainly unrelated.

**Comment by mmitche** [comment]
@AndyAyersMS @janvorli I don't know...what changes went in which could have broken this (to corefx or coreclr)

**Pull request merged**